controllers/datasource_controller.go

```diff
@@ -156,7 +156,7 @@ func (r *DatasourceReconciler) Checkdatasource(ctx context.Context, logger logr.
 	}
 
 	// check datasource
-	if err := ds.Check(ctx, nil); err != nil {
+	if err := ds.Check(ctx, instance.Spec.OSS.DeepCopy()); err != nil {
 		return r.UpdateStatus(ctx, instance, err)
 	}
```

Comment:
**Suggestion:** Handle the error returned by `UpdateStatus` method in the `Checkdatasource` function.
```suggestion
 func (r *DatasourceReconciler) Checkdatasource(ctx context.Context, logger logr.Logger, instance *v1alpha1.Datasource) error {
     // check datasource
     if err := ds.Check(ctx, instance.Spec.OSS.DeepCopy()); err != nil {
         if updateErr := r.UpdateStatus(ctx, instance, err); updateErr != nil {
             logger.Error(updateErr, "Failed to update status")
         }
         return err
     }
 }
```